streams/service/src/main/java/com/hortonworks/streamline/streams/service/SchemaResource.java

```diff
@@ -133,24 +145,37 @@ public Response getKafkaSourceSchema(@PathParam("topicName") String topicName,
         }
     }
 
+    @GET
+    @Path("/{topicName}/{branchName}")
+    @Timed
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getKafkaSourceSchemaForBranch(@PathParam("topicName") String topicName,
```

Comment:
We probably dont need this endpoint if user is forced to select a version from UI. Current behavior is user first selects a topic which populates the version drop down and then once a version is selected the schema is displayed in UI for that version.

Reply:
I have added respective APIs for branching but it seems the current API of "/{topicName}" is also not used. In that case, we can remove this API.